### PR TITLE
DEPS: Remove old pandas compatibility functionality

### LINKED ISSUE #2454: DEPS: Increase minimum pandas version and remove old compatibility functionality
xref: https://github.com/ibis-project/ibis/pull/2450/files#r500471390

We are currently [supporting pandas 0.21+](https://github.com/ibis-project/ibis/blob/master/requirements.txt#L3), which was released in 2017. And we've got some functionality to support moved APIs in `ibis/compat.py`. We can increase the version of pandas, so that compatibility stuff is not needed anymore. Not sure what version of pandas is the minimum that has the new API that doesn't require that compat stuff, but that would be a good one to require as minimum.

### PR DESCRIPTION
**Pull request by EdAbati**
closes #2454

I have removed the pandas imports from `compat.py` and made explicit imports from pandas in the scripts that were using compat.

The only variable that remains in compat is `PY38`, probably it should be moved somewhere else. What do you think?

Please let me know if I can improve this.

Thanks

### CODE REVIEW

File: ibis/compat.py [hunk 1]
```diff
@@ -1,29 +1,4 @@
 """Module for compatible functions."""
-import operator
 import sys
 
-import toolz
```
Comment (jreback) [inline ibis/compat.py:5]: hmm can we completely remove this module?

File: ibis/expr/api.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import dateutil.parser
 import pandas as pd
 import toolz
+from pandas.core.tools.datetimes import to_datetime, to_time
```
Comment (jreback) [inline ibis/expr/api.py:13]: use `pd.to_datetime`, unfortunatly `to_time` is not exposed directly

File: ibis/pandas/client.py [hunk 3]
```diff
@@ -19,7 +20,6 @@
 import ibis.expr.operations as ops
 import ibis.expr.schema as sch
 import ibis.expr.types as ir
-from ibis.compat import CategoricalDtype, DatetimeTZDtype
 from ibis.pandas.core import execute_and_reset
 
 try:
```
Comment (jreback) [inline ibis/pandas/client.py:25]: this can be removed as well (the try/excpet)

**Review by jreback — CHANGES_REQUESTED** [review]

File: ibis/compat.py (hunk 1 above)
Comment (EdAbati) [reply ibis/compat.py:5]: Yes I agree, `PY38` should probably be moved somewhere else.

At the moment I think it is only used for BigQuery. For example in the udf tests:
https://github.com/ibis-project/ibis/blob/e91e635e852308ed9b0d819a83becf983f5e8cbb/ibis/backends/bigquery/tests/udf/test_core.py#L11
https://github.com/ibis-project/ibis/blob/e91e635e852308ed9b0d819a83becf983f5e8cbb/ibis/backends/bigquery/tests/udf/test_find.py#L10
https://github.com/ibis-project/ibis/blob/e91e635e852308ed9b0d819a83becf983f5e8cbb/ibis/backends/bigquery/tests/udf/test_udf_execute.py#L16

And it is also imported in the following `__init__.py` , but (as far as I can tell) only used in a doctstring:
https://github.com/ibis-project/ibis/blob/e91e635e852308ed9b0d819a83becf983f5e8cbb/ibis/backends/bigquery/udf/__init__.py#L9

Where do you think I should move it to?

File: ibis/pandas/client.py (hunk 3 above)
Comment (jreback) [reply ibis/pandas/client.py:25]: if you can do this would be great

File: ibis/expr/api.py (hunk 2 above)
Comment (EdAbati) [reply ibis/expr/api.py:13]: I am not sure if it was you were suggesting, but I've  changed `to_time(value)` with `pd.to_datetime(value).time()`
Is it what you were referring to?

Comment (jreback) [reply ibis/expr/api.py]: hmm that's not what i meant, but if it works would be happier with this (as `to_time`) is not exposed in the `pd` namespace until 1.2 IIRC but i guess it worked.

**Review by jreback — APPROVED** [review]
thanks @EdAbati very nice

**Pull request merged**